src/js/index.js

```diff
@@ -0,0 +1,309 @@
+ `{arguments}.0` in your listener arguments.  The functions which are wrapped like this are:  `goto`, `back`, `forward`,
+ `refresh`, `click`, `type`, `check`, `select`, `scrollTo`, `inject`, and `wait`.
+
+ Nightmare also provides test functions that return a value based on the current state of the browser.  These are also
+ wrapped in invokers that fire an event with the native result.  For these, `{arguments}.0` corresponds to the
+ value that would have be returned by calling the function directly.  These functions are: `exists`, `visible`, `title`,
+ and `url`.
+
+ Nightmare provides two screenshot functions (`screenshot` and `pdf`).  These are wrapped in invokers that fire an
+ event when the screenshot is complete.  The returned value (`{arguments}.0`) represents the path to the generated file.
+
+ In addition to the functions provided by Nightmare, we provide:
+
+    1. `sendKey(int keyCode)`: a function to simulate a keyboard event.  Fires `onKeySent` when complete.
+    2. `getFocusedElement()`: a function to get the object that is currently focused
+    3. `isFocused(selector)`: a function that returns true if the specified selector is currently focused.
+
+  See the tests in this package for examples of how this can all be used together.
+
+ */
+"use strict";
+var fluid = require("infusion");
+var gpii = fluid.registerNamespace("gpii");
+
+fluid.registerNamespace("gpii.tests.browser");
+
+gpii.tests.browser.loadTestingSupport = function () {
+    require("../../tests/js/lib/fixtures");
+    require("../../tests/js/lib/sanity");
+};
+
+var Nightmare = require("nightmare");
+var os        = require("os");
+var path      = require("path");
+
+gpii.tests.browser.init = function (that) {
+    that.nightmare = new Nightmare(that.options.nightmareOptions); // jshint ignore: line
+
+    fluid.each(that.options.eventBindings, function (rawEventOrEvents, fluidEvent) {
+        var rawEvents = fluid.makeArray(rawEventOrEvents);
+        fluid.each(rawEvents, function (rawEvent) {
+            that.nightmare.on(rawEvent, function () {
+                that.events[fluidEvent].fire(fluid.makeArray(arguments));
+            });
+        });
+    });
+
+    that.events.onReady.fire(that);
+};
+
+gpii.tests.browser.execute = function (that, eventName, fnName, args) {
+    var argsArray = fluid.makeArray(args);
+    that.nightmare[fnName].apply(that.nightmare, argsArray).run(function (err, result) {
+        if (err) {
+            that.events.onError.fire(err);
+        } else {
+            that.events[eventName].fire(result);
+        }
+    });
+};
+
+// TODO:  Review and see why the process must be killed.
+gpii.tests.browser.end = function (that) {
+    that.nightmare.end();
+
+    // For some reason the "end" callback does not always execute, so we use an interval to take care of business
+    var millis = 0, intervalTimer = setInterval(function () {
+        millis += that.options.endInterval;
+        if (that.nightmare.ended === true) {
```

Comment:
I assume that is your way of saying the previous syntax was redundant.  I simplified the check.